tbdex/rfq/rfq_test.go

```diff
@@ -81,7 +81,57 @@ func TestRFQ_Sign(t *testing.T) {
 
 func TestRFQ_UnmarshalJSON(t *testing.T) {
 	pfiDID, _ := didjwk.Create()
+	walletDID, _ := didjwk.Create()
+	offeringID, _ := typeid.WithPrefix(offering.Kind)
+
+	r, _ := rfq.Create(
+		walletDID.URI,
+		pfiDID.URI,
+		offeringID.String(),
+		rfq.Payin("100", "STORED_BALANCE"),
+		rfq.Payout("BANK_ACCOUNT"),
+	)
+
+	_ = r.Sign(walletDID)
+
+	bytes, err := json.Marshal(r)
+	assert.NoError(t, err)
+
+	var rfq rfq.RFQ
+	err = rfq.ValidateAndUnmarshalJSON(bytes, false)
+	assert.NoError(t, err)
+}
+
+func TestRFQ_UnmarshalJSON_FailsIncompletePrivateData(t *testing.T) {
```

Comment:
I'm not sure why unmarshalling should fail in this test. `rfq.Create` should output the RFQ with all the private data, no? why are is marshalling producing an error?

Reply:
marshaling is not producing an error, UNmarshaling is. This is testing that unmarshal fails if `strict` is true, which is how I interpreted the method should work based on the docs in `tbdex-js`:

```
/**
   * Verify the presence and integrity of all possible properties in {@link Rfq.privateData}.
   * @throws if there are properties missing in {@link Rfq.privateData} or which do not match the corresponding
   *         hashed property in {@link Rfq.data}
   */
```

Let me know if it should function differently!